### PR TITLE
Cherry-picking - (rm python2 % add python3 .format)

### PR DESCRIPTION
**Pull request by michaelbadcrumble**
Cherry-picking Python2 % syntax from Meson commands.

### CODE REVIEW

File: mesonbuild/build.py [hunk 1]
```diff
@@ -964,7 +963,7 @@ def process_kwargs(self, kwargs, environment):
         if self.gnu_symbol_visibility != '':
             permitted = ['default', 'internal', 'hidden', 'protected', 'inlineshidden']
             if self.gnu_symbol_visibility not in permitted:
-                raise InvalidArguments('GNU symbol visibility arg %s not one of: %s',
+                raise InvalidArguments('GNU symbol visibility arg {} not one of: {}',
                                        self.symbol_visibility, ', '.join(permitted))
```
Comment (jpakkane) [inline mesonbuild/build.py]: This is missing `.format`. TBF the original was broken as well..

Comment (michaelbadcrumble) [reply mesonbuild/build.py]: Added the missing `.format`.

Comment (michaelbadcrumble) [reply mesonbuild/build.py]: 😕 So.  Should there be a test to catch this?

Comment (jpakkane) [reply mesonbuild/build.py]: Given the way Python works, it's not really feasible.

**Pull request merged**